azurelinuxagent/ga/monitor.py

```diff
@@ -424,14 +424,11 @@ def send_telemetry_heartbeat(self):
                 dropped_packets = self.osutil.get_firewall_dropped_packets(self.protocol.get_endpoint())
                 msg = "{0};{1};{2};{3}".format(incarnation, self.counter, self.heartbeat_id, dropped_packets)
 
-                add_event(
-                    name=AGENT_NAME,
-                    version=CURRENT_VERSION,
-                    op=WALAEventOperation.HeartBeat,
-                    is_success=True,
-                    message=msg,
-                    log_event=False)
+                add_event(name=AGENT_NAME, version=CURRENT_VERSION, op=WALAEventOperation.HeartBeat, is_success=True,
+                          message=msg, log_event=False)
 
+                logger.periodic_info(logger.EVERY_HALF_DAY, "[PERIODIC] Incarnation: {0}; ContainerId: {2}",
```

Comment:
Let's add more context to the message (what is it?)

Let's not include the container ID (I'm even  tempted to remove the incarnation from the heartbeat if we are not using it),  we are still working on those calls update_goal_state. If we do the heartbeat on the main thread (extension handling) then I do not have concerns about this.

Reply:
> I'm even tempted to remove the incarnation

since the heartbeat thread is separate from extension handling I think the incarnation here has no meaning since it may be out of sync with the incarnation that we are processing or processed last

Reply:
Yup - I'll add more context.

>If we do the heartbeat on the main thread (extension handling) then I do not have concerns about this.

This sounds like a good idea. I'll add a logger in the ext-handler's run method. This would help us catch cases like the one faced by the AzureData team, where the ext-handler was down but the monitor-handler was sending messages.

> Let's not include the container ID (I'm even tempted to remove the incarnation from the heartbeat if we are not using it), 

I will keep incarnation to match it with WindowsGA, but other info can cause the periodic entry to break (Hash).
```
string message = string.Format("{0};{1};{2}",
    this.Incarnation,       // current goalstate incarantion
    this.HeartBeatCount++,  // current heart beat count 
    Utilities.ProcessId);
```

Reply:
+1 on the ProcessId, I like the idea that if something went wrong and the agent had to restart we would capture that in our logs/telemetry too.

Reply:
The Pid is now populated in the Kusto logs. I haven't added PID in the current HeartBeat, but the logs will tell if a restart happened.